services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java

```diff
@@ -902,6 +932,12 @@ public MapboxDirections build() {
         approaches(formattedApproaches);
       }
 
+      if (walkingOptions() == null) {
+        walkingOptions(WalkingOptions.builder().build());
+      } else if (!DirectionsCriteria.PROFILE_WALKING.equals(profile())) {
+        throw new ServicesException("Walking options are for use with the walking profile");
```

Comment:
Does the backend service return any error in this situation? If so, we should bubble up that error and not thrown an exception at this point.